README.md

```diff
@@ -122,6 +122,239 @@ the components involved in the test:
+
+The read operation is implicit.
+
+##### Managed
+
+The E2E managed tests will spin up the same environment as the
+[Integration](README.md#integration) tests described above and run [CRUD
+(create-read-update-delete)
+checks](https://github.com/kubernetes-sigs/federation-v2/blob/master/test/e2e/crud.go) for
+federated types against that federation. To run:
+
+ - ensure the same binaries are available as described in the
+   [Integration](README.md#integration) section.
+
+To run tests for all types:
+
+```bash
+cd test/e2e
+go test -v
+```
+
+To run tests for a single type:
+
+```bash
+cd test/e2e
+go test -args -v=4 -test.v --ginkgo.focus='"FederatedSecret"'
+```
+
+It may be helpful to use the [delve
+debugger](https://github.com/derekparker/delve) to gain insight into
+the components involved in the test:
+
+```bash
+cd test/e2e
+dlv test -- -v=4 -test.v --ginkgo.focus='"FederatedSecret"'
+```
+
+##### Unmanaged and Hybrid Cluster Setup
+
+The difference between unmanaged and hybrid is that with hybrid, you run
+the federation-v2 controllers in-process as part of executing the test. This
+helps in running a debugger to debug anything in the controllers. On the other
+hand with unmanaged, the controllers are already running in the K8s cluster.
+Both methods require the clusters to have already been joined.
+
+###### Create Clusters
+
+The quickest way to set up clusters for use with unmanaged and hybrid modes is
+to use [minikube](https://kubernetes.io/docs/getting-started-guides/minikube/).
+Once you have minikube installed run:
+
+```bash
+minikube start -p clusterA
+```
+
+###### Deploy the Cluster Registry
+
+Get the latest version of the [Cluster
+Registry](https://github.com/kubernetes/cluster-registry/releases) and run the
+following:
+
+```bash
+crinit aggregated init mycr --host-cluster-context=clusterA
+```
+
+You can also specify your own cluster registry image using the `--image` flag.
+
+###### Deploy Federation
+
+First you'll need to create the namespace to deploy the federation into:
+
+```bash
+kubectl create ns federation
+```
+
+Then run `apiserver-boot` to deploy. This will build the federated API server
+and controller manager, build an image containing them, push the image to the
+requested registry, generate a k8s YAML config `config/apiserver.yaml`
+with the name and namespace requested, and deploy the config to the
+cluster specified in your kubeconfig current-context.
+
+```bash
+apiserver-boot run in-cluster --name federation --namespace federation \
+    --image <containerregistry>/<username>/<imagename>:<tagname> \
+    --controller-args="-logtostderr,-v=4"
+```
+
+You will most likely need to increase the limit on the amount of memory
+required by the API server as there isn't a way to request it via
+`apiserver-boot` at the moment. Otherwise the federation pod may be terminated
+with `OOMKilled`. In order to do that you will need to modify the deployment:
+
+```bash
```

Comment:
Consider supplying a 'kubectl patch` invocation instead: 

https://github.com/kubernetes-sigs/federation-v2/pull/57/files#diff-1ed12d3b5689c7a45b1c21e93e23e3d3R69

Reply:
Yes, thanks! I knew there'd be a way to do that but I wasn't familiar with it and needed to just capture the issue and move on. Similarly, we should include a patch command to delete the controller in the hybrid setup too. Ideally, I think this should just be part of the deployment config committed to the tree (see comment above).

Reply:
Done.